core/domain/html_validation_service.py

```diff
@@ -932,21 +924,21 @@ def _modify_image_filename(exploration_fs, soup):
     for image in soup.findAll(name='oppia-noninteractive-image'):
         if (not image.has_attr('filepath-with-value') or
                 image['filepath-with-value'] == '' or
-                image['filepath-with-value'] == '\\"""\\"'):
+                image['filepath-with-value'] == '""' or
+                image['filepath-with-value'] == '&quot;&quot;'):
             image.decompose()
             continue
 
         try:
             filename = (
-                json.loads(unescape_html(image['filepath-with-value']
-                                         .replace('\\"', ''))))
+                json.loads(unescape_html(image['filepath-with-value'])))
             escaped_filename = escape_html(
                 json.dumps(
                     get_filename_with_dimensions(exploration_fs, filename)))
-            image['filepath-with-value'] = '\\"' + escaped_filename + '\\"'
+            image['filepath-with-value'] = escaped_filename
         except Exception as e:
             (entity_type, entity_id) = (
-                exploration_fs.impl.assets_path.split('/'))
+                exploration_fs.impl.assets_path.split('/')[:2])
```

Comment:
Interesting -- was this just wrong before?

(How come ... if it was covered by tests?)

Reply:
`exploration_fs.impl.assets_path.split('/')` outputs a list containing 3 strings. Earlier we were accessing only elements 1 and 2.
eg.
```python
entity_type = exploration_fs.impl.assets_path.split('/')[0]
entity_id = exploration_fs.impl.assets_path.split('/')[1]
``` 

Now that we simplified this to store the output to a tuple, it would require a 3rd unused variable. To avoid that I've added `[:2]`.

Reply:
Ah OK. I think it's better if you do `(entity_type, entity_id, _) = exploration_fs.impl.asasets_path.split('/')`.